crawl-ref/source/mon-cast.cc

```diff
@@ -4937,6 +4938,79 @@ dice_def resonance_strike_base_damage(const monster &mons)
     return dice_def(3, mons.spell_hd(SPELL_RESONANCE_STRIKE));
 }
 
+static void _sheep_message(int num_sheep, int sleep_pow, actor& foe)
+{
+    string message;
+
+    // Determine messaging based on sleep strength.
+    if (sleep_pow >= 125)
+        message = "You are overwhelmed by glittering dream dust!";
+    else if (sleep_pow >= 75)
+        message = "The dream sheep are wreathed in dream dust.";
+    else if (sleep_pow >= 25)
+    {
+        message = make_stringf("The dream sheep shake%s wool and sparkle%s.",
+                               num_sheep == 1 ? "s its" : " their",
+                               num_sheep == 1 ? "s": "");
+    }
+    else // if sleep fails
+    {
+        message = make_stringf("The dream sheep ruffle%s wool and motes of "
+                               "dream dust sparkle, to no effect.",
+                               num_sheep == 1 ? "s its" : " their");
+    }
+
+    if (!foe.is_player()) // Messaging for non-player targets
+    {
+        if (you.see_cell(foe.pos()) && sleep_pow)
+        {
+            mprf(foe.as_monster()->friendly() ? MSGCH_FRIEND_SPELL
+                                              : MSGCH_MONSTER_SPELL,
+                 "As the sheep sparkle%s and sway%s, %s falls asleep.",
+                 num_sheep == 1 ? "s": "",
+                 num_sheep == 1 ? "s": "",
+                 foe.name(DESC_THE).c_str());
+        }
+        else // if dust strength failure for non-player
+        {
+            mprf(foe.as_monster()->friendly() ? MSGCH_FRIEND_SPELL
+                                              : MSGCH_MONSTER_SPELL,
+                 "The dream sheep attempt%s to lull %s to sleep.",
+                 num_sheep == 1 ? "s" : "",
+                 foe.name(DESC_THE).c_str());
+            mprf("%s is unaffected.", foe.name(DESC_THE).c_str());
+        }
+    }
+    else
+    {
+        mprf(MSGCH_MONSTER_SPELL, "%s%s", message.c_str(),
+             sleep_pow ? " You feel drowsy..." : "");
+    }
+}
+
+static void _dream_sheep_sleep(monster& mons, actor& foe)
+{
+    // Shepherd the dream sheep.
+    vector<monster*> dream_herd;
```

Comment:
Why collect a vector of all of them, if all you care about is how many there are? Just start an int at 0 and increment it for each sheep.